### ISSUE TITLE
Failed to resolve module: xo. Ensure that xo is installed.

### ISSUE THREAD

**Issue by Yash-Singh1** [xojs/vscode-linter-xo#100]
Some of my repositories don't use `xo` and every second that I type some javascript code, a popup appears saying: `Failed to resolve module: xo. Ensure that xo is installed.`

![Screenshot from 2021-10-14 18-46-38](https://user-images.githubusercontent.com/53054099/137419200-1503f47d-fa1b-4934-b049-fb5a1ff98a34.png)

Maybe, the extension will not show the popup if it is not installed because not all repositories use `xo`.

Thanks for making this vscode extension!

**Comment by Spence-S** [comment]
You can disable the extension per workspace in vscode. 

Perhaps this is too aggressive. I'll leave this open and see if others find it too annoying. I will just have it report this particular error in the in the extension tab only.

**Comment by tbrolin** [comment]
Yesterday I reopened my project and started to get the failed to resolve module xo error. My workspace root is not a npm package and only one subfolder is using xo, with xo configuration in the package.json file. The plugin has worked fine up until now...

**Comment by Spence-S** [comment]
I went ahead and published an update so that particular error message should only be shown once and then it will continue to try to resolve xo in the background, without showing an error, until you install it. 

I may consider removing this message altogether in the future but I feel it could be helpful to new users. Hopefully this is a good middle ground.

> My workspace root is not a npm package and only one subfolder is using xo, with xo configuration in the package.json file.

Not sure what your workflow is here, but 3.2 does not have proper support for "multi-root/multi-folder" workspaces. 3.3+ now can resolve different versions of xo for each folder and apply them properly. It sounds like you were possibly using the a work around that no longer applies now that full support is in 3.3

**Comment by darkred** [comment]
I also find this message too annoying and aggressive.

So, I'd like to offer an approach I came up with: to check the current folder tree's package.json, and 
- if there is a `xo` entry in package.json, e.g. https://github.com/refined-github/refined-github/blob/8ed6674f0935ff23dbb4ea81f2f327b4c748405a/package.json#L125
but `xo` is not installed (i.e. the user hasn't run `npm install`), then display the message .
- if there's no `xo` entry, e.g. of my repo https://github.com/darkred/Userscripts/blob/master/package.json
then never show the message at all.

**Comment by Spence-S** [comment]
@darkred @sachinraja - do you feel it is still aggressive even as it is now - showing only 1 time per open instance of VScode?

@darkred I like that idea - I will implement it that way. - however, I won't make it depend on xo being in package.json, it will look attempt to resolve xo and lint no matter what, but if xo isn't present in deps - it will fail silently.

@sachinraja xo does not need any config files to work. It requires 0 config so searching for a config like the eslint extension does is pointless.

**Comment by sachinraja** [comment]
I do feel that is still aggressive as the error shows every time I open vscode to a project without xo.

**Comment by darkred** [comment]
> do you feel it is still aggressive even as it is now - showing only 1 time per open instance of VScode?

Yes, because e.g. my repo doesn't use xo, and I shouldn't get the message at all.
And, in general, XO, being an ESLint wrapper, regardless of how much useful it is, it's less commonly used than ESLint alone.

**Comment by sachinraja** [comment]
Thank you for listening to our feedback @Spence-S! Really appreciate your work on this extension
